Adds a Custom Widget Use Case

mjelzon opened the pull request:
also adds a CustomAlertWidgetViewController and CustomWidgetTimer

Commit:
adds a Custom Widget use case to the home page

adds a CustomAlertWidgetViewController and CustomWidgetTimer

mmoloksher commented:
@mjelzon i believe you have the timer going backwards, it should go left to right

Commit:
Changes the direction of the timer animation

mmoloksher commented:
Any reason you decided to separate CustomAlertViewController and CustomAlertView? I feel like it’s unnecessary and makes the custom code consist of two files instead of one. Not a blocker though

mjelzon commented:
> Any reason you decided to separate CustomAlertViewController and CustomAlertView? I feel like it’s unnecessary and makes the custom code consist of two files instead of one. Not a blocker though

mjelzon commented:
I thought it'd be best practice to have a dumb view and to use the viewcontroller for business logic. With a dumb view it can easily be reused outside the context of widgets (WidgetModel).

Something notable is that the CustomAlertView doesn't depend on EngagementSDK and is 100% UIKit code. This could help showcase how much an integrator should already know how to do vs. what they have to learn about the EngagementSDK.

mmoloksher commented on Sample App/Custom Widgets/CustomWidgetTimer.swift:
```diff
@@ -0,0 +1,62 @@
+
+    var progressView: UIView = {
+        let view = UIView()
+        view.backgroundColor = UIColor(red: 250/255, green: 200/255, blue: 60/255, alpha: 1)
```
btw this shows up black for me, `UIColor(red: 250, green: 200, blue: 60, alpha: 1)` works

mjelzon replied:
Thats because we have this extension on the TestApp

```
extension UIColor {
    convenience init(red: Int, green: Int, blue: Int, alpha: CGFloat = 1.0) {
        let newRed = CGFloat(red) / 255

        let newGreen = CGFloat(green) / 255

        let newBlue = CGFloat(blue) / 255

        self.init(red: newRed, green: newGreen, blue: newBlue, alpha: alpha)
    }
}
```

We should make sure to use the native init for sample app

mmoloksher replied:
Good catch

mmoloksher reviewed (commented):
The color of the progress bar shows up black for me, `UIColor(red: 250, green: 200, blue: 60, alpha: 1)` works

Commit:
Adds the WidgetViewController in the custom widget use case

Adds the bar timer to the CustomAlertWidgetView
Adds CustomAlertWidgetViewController inherits from Widget
